src/java/org/apache/cassandra/db/rows/Cell.java

```diff
@@ -108,7 +127,7 @@ public ByteBuffer buffer()
      * @return the cell local deletion time, or {@code NO_DELETION_TIME} if the cell is neither
      * a tombstone nor an expiring one.
      */
-    public abstract int localDeletionTime();
+    public abstract long localDeletionTime();
```

Comment:
Every implementation of this method contains a call to `Cell.deletionTimeGuavaUintToLong`. Instead, we could have an additional `protected` method that returns the unsigned int, and put the call to `Cell.deletionTimeGuavaUintToLong` here:
```java
public long localDeletionTime()
{
    return deletionTimeGuavaUintToLong(localDeletionTimeAsUnsignedInt());
}

protected abstract int localDeletionTimeAsUnsignedInt();
```
This simplifies a bit the implementations and might skip a conversion on the first constructor of `NativeCell`:
```java
public NativeCell(NativeAllocator allocator,
                  OpOrder.Group writeOp,
                  Cell<?> cell)
{
    this(allocator,
         writeOp,
         cell.column(),
         cell.timestamp(),
         cell.ttl(),
         cell.localDeletionTimeAsUnsignedInt(),
         cell.buffer(),
         cell.path());
}

public NativeCell(NativeAllocator allocator,
                  OpOrder.Group writeOp,
                  ColumnMetadata column,
                  long timestamp,
                  int ttl,
                  long localDeletionTime,
                  ByteBuffer value,
                  CellPath path)
{
    this(allocator, writeOp, column, timestamp, ttl, deletionTimeLongToGuavaUint(lo
}

private NativeCell(NativeAllocator allocator,
                   OpOrder.Group writeOp,
                   ColumnMetadata column,
                   long timestamp,
                   int ttl,
                   int localDeletionTimeUint,
                   ByteBuffer value,
                   CellPath path)
{
...
}
```

Reply:
So the implementations of `Cell` will now _still_ have to implement one method? I don't see the gain unless I am missing sthg.

Adding a `NativeCell` int friendly constructor sound like a good idea.

Reply:
All the implementations of `Cell#localDeletionTime` are wrapped in a call to `Cell.deletionTimeGuavaUintToLong`:
* ArrayCell: `return deletionTimeGuavaUintToLong(localDeletionTimeGuavaUint);`
* BufferCell: `return deletionTimeGuavaUintToLong(localDeletionTimeGuavaUint);`
* NativeCell: `return deletionTimeGuavaUintToLong(MemoryUtil.getInt(peer + DELETION));`

If we put that call in a common implementation of `Cell#localDeletionTime` and add an abstract `localDeletionTimeAsUnsignedInt ` method, the implementations would be simplified to:
* ArrayCell: `return localDeletionTimeGuavaUint;`
* BufferCell: `return localDeletionTimeGuavaUint;`
* NativeCell: `return MemoryUtil.getInt(peer + DELETION);`

More importantly, we could use that `Cell#localDeletionTimeAsUnsignedInt` method to avoid pointless conversions in `NativeCell(NativeAllocator, OpOrder.Group, Cell<?>)`. At the moment that method calls `Cell#localDeletionTime`, which does a uint->long conversion. That long is passed to the next constructor, that immediately does the reverse long->uint conversion. We can avoid that with the `localDeletionTimeAsUnsignedInt` method show in the code snippet shown in my previous comment.

Reply:
Also, `NativeCell#withUpdatedTimestampAndLocalDeletionTime`, `NativeCell#withUpdatedColumn` and `NativeCell#withSkippedValue` could also benefit from that `Cell#localDeletionTimeAsUnsignedInt` method to avoid the uint->long->uint conversion that the currently do.

Reply:
Right let's avoid conversions as much as possible. Note: `withUpdatedTimestampAndLocalDeletionTime()` passes a param for ldt so it doesn't apply there iiuc.